src/components/ui/dialog-modal.tsx

```diff
@@ -0,0 +1,225 @@
+import * as React from "react"
+import { MdClose } from "react-icons/md"
+import { tv, type VariantProps } from "tailwind-variants"
+import * as DialogPrimitive from "@radix-ui/react-dialog"
+
+import { cn } from "@/lib/utils/cn"
+
+import { Button } from "./buttons/Button"
+import { Center, Flex } from "./flex"
+
+const dialogVariant = tv({
+  slots: {
+    content:
+      /**
+       * `size-[calc(-2rem_+_100%)]` provides 16px x- and y-spacing
+       * around the container instead of using `margin` to
+       * avoid positioning side-effects.
+       * (The container is fixed-positioned)
+       *
+       * Credit: Chakra v2 takes a similar approach for modal
+       */
```

Comment:
Comment regarding using of `calc` with height and width instead of `margin` to have spacing around the container when the viewport gets too small.